consensus/hotstuff/pacemaker/pacemaker_test.go

```diff
@@ -30,456 +32,188 @@ func expectedTimerInfo(view uint64, mode model.TimeoutMode) interface{} {
 		})
 }
 
-func expectedTimeoutInfo(view uint64, mode model.TimeoutMode) interface{} {
-	return mock.MatchedBy(
-		func(timerInfo *model.TimerInfo) bool {
-			return timerInfo.View == view && timerInfo.Mode == mode
-		})
+func TestActivePaceMaker(t *testing.T) {
+	suite.Run(t, new(ActivePaceMakerTestSuite))
 }
 
-func initPaceMaker(t *testing.T, view uint64) (hotstuff.PaceMaker, *mocks.Consumer) {
-	notifier := &mocks.Consumer{}
+type ActivePaceMakerTestSuite struct {
+	suite.Suite
+
+	livenessData *hotstuff.LivenessData
+	notifier     *mocks.Consumer
+	persist      *mocks.Persister
+	paceMaker    *ActivePaceMaker
+}
+
+func (s *ActivePaceMakerTestSuite) SetupTest() {
+	s.notifier = &mocks.Consumer{}
+	s.persist = &mocks.Persister{}
+
 	tc, err := timeout.NewConfig(
 		time.Duration(startRepTimeout*1e6),
 		time.Duration(minRepTimeout*1e6),
 		voteTimeoutFraction,
 		multiplicativeIncrease,
 		multiplicativeDecrease,
 		0)
-	if err != nil {
-		t.Fail()
-	}
-	pm, err := New(view, timeout.NewController(tc), notifier)
-	if err != nil {
-		t.Fail()
+	require.NoError(s.T(), err)
+
+	s.livenessData = &hotstuff.LivenessData{
+		CurrentView: 3,
+		LastViewTC:  nil,
+		NewestQC:    helper.MakeQC(helper.WithQCView(2)),
 	}
-	notifier.On("OnStartingTimeout", expectedTimerInfo(view, model.ReplicaTimeout)).Return().Once()
-	pm.Start()
-	return pm, notifier
+
+	s.persist.On("GetLivenessData").Return(s.livenessData, nil).Once()
+
+	s.paceMaker, err = New(timeout.NewController(tc), s.notifier, s.persist)
+	require.NoError(s.T(), err)
+
+	s.notifier.On("OnStartingTimeout", expectedTimerInfo(s.livenessData.CurrentView, model.ReplicaTimeout)).Return().Once()
+
+	s.paceMaker.Start()
 }
 
 func QC(view uint64) *flow.QuorumCertificate {
 	return &flow.QuorumCertificate{View: view}
 }
 
-//func makeBlock(qcView, blockView uint64) *model.Block {
-//	return &model.Block{View: blockView, QC: QC(qcView)}
-//}
+func LivenessData(qc *flow.QuorumCertificate) *hotstuff.LivenessData {
+	return &hotstuff.LivenessData{
+		CurrentView: qc.View + 1,
+		LastViewTC:  nil,
+		NewestQC:    qc,
+	}
+}
 
-// Test_SkipIncreaseViewThroughQC tests that PaceMaker increases View when receiving QC,
+// TestProcessQC_SkipIncreaseViewThroughQC tests that ActivePaceMaker increases view when receiving QC,
 // if applicable, by skipping views
-func Test_SkipIncreaseViewThroughQC(t *testing.T) {
-	pm, notifier := initPaceMaker(t, 3)
-
+func (s *ActivePaceMakerTestSuite) TestProcessQC_SkipIncreaseViewThroughQC() {
 	qc := QC(3)
```

Comment:
```suggestion
	qc := QC(s.livenessData.CurrentView)
```
There are a lot of numeric literals used here and in the other test cases - it would be nice if these values were computed based on the test suite configuration, instead of using constants:
* the test will work without changes if the initial state changes
* it's also clearer to a reader, `QC(curView)` tells me that the QC is for the current view - I don't have to refer back to the config